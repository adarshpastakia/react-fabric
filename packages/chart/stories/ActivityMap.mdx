import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { ActivityMap } from "../src";
import * as ActivityMapStories from "./ActivityMap.stories";

<Meta of={ActivityMapStories} />

<Title>Activity Map</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Activity Map</Subheading>
  </div>
</Anchor>

<Description of={ActivityMap} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={ActivityMapStories.Example} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="data">
  <Subheading>Data Format</Subheading>
</Anchor>

```ts
// Hours of day
{
    time: "day-hour",
    data:[
        // day 1
        [[hour, day, count],...],
        // day 2
        [[hour, day, count],...]
    ]
}

// Days of month
{
    time: "month-day",
    data:[
        // month 1
        [[day, month, count],...],
        // month 2
        [[day, month, count],...]
    ]
}
```
