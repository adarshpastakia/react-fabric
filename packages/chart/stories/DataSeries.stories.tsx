/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { faker } from "@faker-js/faker";
import { Button, Panel } from "@react-fabric/core";
import type { Meta, StoryObj } from "@storybook/react";
import { useCallback, useEffect, useState } from "react";
import { DataSeries } from "../src";

const meta: Meta<typeof DataSeries> = {
  component: DataSeries,
  title: "@charts/DataSeries",
  parameters: {
    layout: "centered",
    controls: { exclude: "children" },
  },
};

export default meta;
type Story = StoryObj<typeof DataSeries>;

export const Example: Story = {
  render: (args) => {
    const [data, setData] = useState<AnyObject>({});
    const loadData = useCallback(() => {
      const categories = ["Jan", "Feb", "Mar", "Apr", "May", "Jun"];
      setData({
        categoryAxisName: "Months",
        valueAxisName: "Items",
        categories,
        data: [
          {
            id: faker.string.alpha(5),
            label: faker.animal.dog(),
            values: categories.map(() =>
              faker.number.int({ min: 100, max: 500 }),
            ),
          },
          {
            id: faker.string.alpha(5),
            label: faker.animal.dog(),
            values: categories.map(() =>
              faker.number.int({ min: 100, max: 500 }),
            ),
          },
          {
            id: faker.string.alpha(5),
            label: faker.animal.dog(),
            values: categories.map(() =>
              faker.number.int({ min: 100, max: 500 }),
            ),
          },
        ],
      });
    }, []);
    useEffect(() => {
      loadData();
    }, []);
    return (
      <Panel
        width="48rem"
        height="24rem"
        title="Data Series chart"
        expandable
        actions={
          <Button
            aria-label="loadData"
            variant="link"
            icon="mdi mdi-refresh"
            onClick={loadData}
          />
        }
      >
        <DataSeries {...args} {...data} />
      </Panel>
    );
  },
  args: {},
};
