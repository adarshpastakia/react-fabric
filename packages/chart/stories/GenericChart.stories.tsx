/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { Button, Panel, ToggleButtonGroup } from "@react-fabric/core";
import type { Meta, StoryObj } from "@storybook/react";
import React, { useEffect, useState } from "react";
import { GenericChart } from "../src";
import coffeeJson from "./coffee";
import sankeyJson from "./sankey";

const meta: Meta<typeof GenericChart> = {
  component: GenericChart,
  title: "@charts/GenericChart",
  parameters: {
    layout: "centered",
    controls: { exclude: "children" },
  },
};

export default meta;
type Story = StoryObj<typeof GenericChart>;

export const TreeMap: Story = {
  render: (args) => {
    const [type, setType] = useState("treemap");
    const [series, setSeries] = useState<AnyObject>({});

    useEffect(() => {
      if (type === "treemap") {
        setSeries({
          id: "coffee-list",
          name: "Coffee list",
          type: "treemap",
          data: coffeeJson,
          universalTransition: true,
          leafDepth: 2,
          upperLabel: {
            show: true,
            height: 30,
          },
          levels: [
            {
              itemStyle: {
                borderColor: "#777",
                borderWidth: 0,
                gapWidth: 1,
              },
              upperLabel: {
                show: false,
              },
            },
            {
              colorSaturation: [0.35, 0.5],
              itemStyle: {
                borderWidth: 5,
                gapWidth: 1,
                borderColorSaturation: 0.6,
              },
            },
          ],
        });
      }
      if (type === "sunburst") {
        setSeries({
          id: "coffee-list",
          name: "Coffee list",
          type: "sunburst",
          radius: ["20%", "90%"],
          data: coffeeJson,
          universalTransition: true,
          levels: [
            {},
            {
              label: {
                rotate: "tangential",
              },
            },
            {
              r0: "35%",
              r: "70%",
              label: {
                align: "right",
              },
            },
            {
              label: {
                position: "outside",
              },
            },
          ],
        });
      }
    }, [type]);
    return (
      <Panel width="48rem" height="24rem" title="Generic chart" expandable>
        <GenericChart
          {...args}
          series={[series]}
          legend={{ show: false }}
          actions={
            <ToggleButtonGroup value={type} onChange={setType}>
              <Button
                value="treemap"
                icon="mdi mdi-view-dashboard-variant"
                aria-label="tree"
              />
              <Button
                value="sunburst"
                icon="mdi mdi-circle-double"
                aria-label="burst"
              />
            </ToggleButtonGroup>
          }
        />
      </Panel>
    );
  },
  args: {},
};

export const Sankey: Story = {
  render: (args) => {
    const [series, setSeries] = useState<AnyObject>({});

    useEffect(() => {
      setSeries({
        type: "sankey",
        emphasis: {
          focus: "adjacency",
        },
        nodeAlign: "right",
        data: sankeyJson.nodes,
        links: sankeyJson.links,
        lineStyle: {
          color: "source",
          curveness: 0.5,
        },
      });
    }, []);
    return (
      <Panel width="48rem" height="24rem" title="Generic chart" expandable>
        <GenericChart {...args} series={[series]} legend={{ show: false }} />
      </Panel>
    );
  },
  args: {},
};
