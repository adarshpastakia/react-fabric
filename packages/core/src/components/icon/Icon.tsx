/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { isString, isSvgPath } from "@react-fabric/utilities";
import classNames from "classnames";
import { useMemoDebugger } from "../../hooks/useEffectDebugger";
import {
  type AriaProps,
  type ColorType,
  type CssProp,
  type RefProp,
  type SizeType,
  type TestProps,
} from "../../types";
import { getBgClass, getColor, getColorClass } from "../../utils";
import classes from "./Icon.module.css";

export interface IconProps extends CssProp, AriaProps, RefProp, TestProps {
  /**
   * svg path / webfont className / 1-4 letter text
   */
  icon: string;
  /**
   * background color (CSS color / tailwind color)
   */
  bg?: ColorType | string;
  /**
   * icon color (CSS color / tailwind color)
   */
  color?: ColorType | string;
  /**
   * flip icon in rtl
   */
  rtlFlip?: boolean;
  /**
   * icon size
   */
  size?: SizeType | number | string;
  /**
   * viewbox for svg path (default 24)
   */
  viewBox?: string;
  /**
   * rounded icon
   */
  rounded?: boolean;
  /**
   * apply animation
   */
  animate?: "spin" | "pulse" | "bounce";
  /**
   * click handler
   */
  onClick?: React.MouseEventHandler;
}

const SvgTextSize = ["", ".875em", ".525em", ".35em", ".25em"];
const SizeMap: KeyValue<string> = {
  xs: "0.625rem",
  sm: "0.875rem",
  md: "1.125rem",
  lg: "1.5rem",
  xl: "2rem",
};

/**
 * An icon is a display component for svg icon path / image
 */
export const Icon = ({
  ref,
  icon,
  bg,
  color,
  size = "",
  rtlFlip,
  animate,
  rounded,
  viewBox = "0 0 24 24",
  className,
  onClick,
  ...rest
}: IconProps) => {
  /** ***************** style map *******************/
  const styles = useMemoDebugger(
    () => {
      const s: KeyValue = {};
      if (bg) {
        s.backgroundColor = getColor(bg);
      }
      if (color) {
        s.color = getColor(color);
      }
      if (size && !(size in SizeMap)) {
        s.fontSize = size;
      }
      if (size && size in SizeMap) {
        s.fontSize = SizeMap[size];
      }
      return s;
    },
    [bg, color, size],
    "Icon styles",
  );

  /** ***************** check if icon is svg path *******************/
  const isSvg = useMemoDebugger(
    () => {
      return isSvgPath(icon);
    },
    [icon],
    "Icon isSvg",
  );

  /** ***************** render icon *******************/
  const iconEl = useMemoDebugger(
    () => {
      if (!isString(icon)) {
        throw Error("Invalid icon expected string");
      }
      return isSvg ? (
        <svg viewBox={viewBox}>
          <path fill="currentColor" d={icon.toString()} />
        </svg>
      ) : icon?.toString().length <= 4 ? (
        <svg role="img">
          <text
            x="50%"
            y="50%"
            dy=".1em"
            dominantBaseline="middle"
            textAnchor="middle"
            style={{ fontSize: SvgTextSize[icon.length] ?? "1em" }}
          >
            {icon}
          </text>
        </svg>
      ) : (
        <i className={`${icon}`} />
      );
    },
    [icon, isSvg, viewBox],
    "Icon el",
  );

  return (
    <dfn
      {...rest}
      ref={ref}
      role="presentation"
      className={classNames(
        classes.icon,
        className,
        "select-none overflow-hidden inline-block box-content leading-none not-italic text-center",
        bg && getBgClass(bg),
        color && getColorClass(color),
        !rounded && "rounded",
        rounded && "rounded-full",
        animate && `animate-${animate}`,
      )}
      onClick={onClick}
      data-flip={rtlFlip}
      style={styles}
      data-clickable={!!onClick}
    >
      {iconEl}
    </dfn>
  );
};
