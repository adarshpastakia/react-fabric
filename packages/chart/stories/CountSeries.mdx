import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { CountSeries } from "../src";
import * as CountSeriesStories from "./CountSeries.stories";

<Meta of={CountSeriesStories} />

<Title>Count Series</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Count Series</Subheading>
  </div>
</Anchor>

<Description of={CountSeries} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={CountSeriesStories.Example} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="data">
  <Subheading>Data Format</Subheading>
</Anchor>

```ts
// Single list of count values
[{ id, label, count },...];
```
