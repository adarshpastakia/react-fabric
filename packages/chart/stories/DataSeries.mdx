import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { DataSeries } from "../src";
import * as DataSeriesStories from "./DataSeries.stories";

<Meta of={DataSeriesStories} />

<Title>Data Series</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Data Series</Subheading>
  </div>
</Anchor>

<Description of={DataSeries} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={DataSeriesStories.Example} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="data">
  <Subheading>Data Format</Subheading>
</Anchor>

```ts
// List of data series
[{ id, label, values: [...] },...];
```
