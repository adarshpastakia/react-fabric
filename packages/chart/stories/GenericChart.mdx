import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { GenericChart } from "../src";
import * as GenericChartStories from "./GenericChart.stories";

<Meta of={GenericChartStories} />

<Title>Generic Chart</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Tree Map / Sunburst</Subheading>
  </div>
</Anchor>

<Description of={GenericChart} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={GenericChartStories.TreeMap} />
</Collapsable>

<Anchor storyId="sankey">
  <Subheading>Sankey</Subheading>
</Anchor>
<Description of={GenericChartStories.Sankey} />
<Canvas withToolbar of={GenericChartStories.Sankey} />
<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={GenericChartStories.Sankey} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />
