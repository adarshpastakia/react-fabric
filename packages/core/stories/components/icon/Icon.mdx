import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { Icon, Collapsable } from "../../../src";
import * as IconStories from "./Icon.stories";

<Meta of={IconStories} />

<Title>Icon</Title>

<Anchor storyId="icon">
  <div className="hidden-anchor">
    <Subheading>Icon</Subheading>
  </div>
</Anchor>

<Description of={Icon} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={IconStories._Icon} />
</Collapsable>

<Anchor storyId="colors">
  <Subheading>Colors</Subheading>
</Anchor>
<Description of={IconStories.Colors} />
<Canvas withToolbar of={IconStories.Colors} />
<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={IconStories.Colors} />
</Collapsable>

<Anchor storyId="sizes">
  <Subheading>Sizes</Subheading>
</Anchor>
<Description of={IconStories.Sizes} />
<Canvas withToolbar of={IconStories.Sizes} />
<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={IconStories.Sizes} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="sample">
  <Subheading>Example</Subheading>
</Anchor>

```tsx
import { Icon } from "@react-fabric/core";

export const Example = () => {
  return <Icon icon="svg path / image url" />;
};
```
