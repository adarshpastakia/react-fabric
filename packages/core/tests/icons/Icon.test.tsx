/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { composeStories } from "@storybook/react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "../../stories/components/icon/Icon.stories";

const { Tester } = composeStories(stories);

describe("Icons", () => {
  it("should render icon", () => {
    const fragment = render(
      <Tester icon="mdi mdi-alien" color="lilac" bg="white" size="md" />,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render rounded icon", () => {
    const fragment = render(
      <Tester icon="mdi mdi-alien" size="2rem" rounded animate="spin" />,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render text icon", () => {
    const fragment = render(<Tester icon="ABC" />);
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should throw error", () => {
    expect(() => render(<Tester />)).toThrow("Invalid icon expected string");
  });
});
