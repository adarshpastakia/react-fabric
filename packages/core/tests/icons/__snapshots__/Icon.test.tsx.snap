// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons should render icon 1`] = `"<div tabindex="0"><div style="display: contents;"><div class="flex gap-2 items-center p-4"><dfn data-ref="icon" role="presentation" class="fabric-icon select-none overflow-hidden box-content leading-none not-italic text-center rounded" style="font-size: 1.125rem;" data-clickable="false" data-inner-clickable="false"><i class="mdi mdi-alien"></i></dfn></div></div></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div>"`;

exports[`Icons should render rounded icon 1`] = `"<div tabindex="0"></div><div tabindex="0"><div style="display: contents;"><div class="flex gap-2 items-center p-4"><dfn data-ref="icon" role="presentation" class="fabric-icon select-none overflow-hidden box-content leading-none not-italic text-center rounded-full animate-spin" style="font-size: 2rem;" data-clickable="false" data-inner-clickable="false"><i class="mdi mdi-alien"></i></dfn></div></div></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div>"`;

exports[`Icons should render text icon 1`] = `"<div tabindex="0"></div><div tabindex="0"></div><div tabindex="0"><div style="display: contents;"><div class="flex gap-2 items-center p-4"><dfn data-ref="icon" role="presentation" class="fabric-icon select-none overflow-hidden box-content leading-none not-italic text-center rounded" data-clickable="false" data-inner-clickable="false"><svg role="img"><text x="50%" y="50%" dy=".1em" dominant-baseline="middle" text-anchor="middle" style="font-size: .35em;">ABC</text></svg></dfn></div></div></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div>"`;
